Updates from cam6_4_070 to cam6_4_085

gold2718 opened the pull request:
- Updated Oslo Aero shadow files (src_cam) plus aero_model.F90 to account for modifications in ESCOMP/CAM between cam6_4_070 and cam6_4_085.
- vertical_diffusion.F90 is still using `pmam_cnst_idx` to decide which constituents to skip for vertical diffusion. We probably want to move to the new `ndropmixed` constituent property. This property is used by CLUBB to decide which transport tendencies to not apply:

> !  Droplet number is transported in dropmixnuc, therefore we
> !  do NOT want CLUBB to apply transport tendencies to avoid double
> !  counting.  Else, we apply tendencies.

Do do this, I have added the `ndropmixed` property to the following constituents, without modifying vertical_diffusion (for now):
```
SO4_NA    SO4_A1    SO4_A2    SO4_AC
SO4_PR    BC_N      BC_AX     BC_NI 
BC_A      BC_AI     BC_AC     OM_NI 
OM_AI     OM_AC     DST_A2    DST_A3
SS_A1     SS_A2     SS_A3     SOA_NA
SOA_A1
```

Commit:
Updates from cam6_4_070 to cam6_4_085

Updated Oslo Aero shadow files (src_cam) plus aero_model.F90 to
account for modifications in ESCOMP/CAM between cam6_4_070 and
cam6_4_085.
- vertical_diffusion.F90 is still using pmam_cnst_idx to decide
  which constituents to skip for vertical diffusion. We probably
  want to move to the new ndropmixed constituent property.
  This property is used by CLUBB to decide which transport tendencies to
  not apply:

  !  Droplet number is transported in dropmixnuc, therefore we
  !  do NOT want CLUBB to apply transport tendencies to avoid double
  !  counting.  Else, we apply tendencies.

gold2718 commented:
>> 
> Is this already covered by the original commit or is it an update?

It is (will be) a separate commit as I think it will change answers and I wanted to be able to track it separately if necessary. My initial attempt did not work but I'm updating everything (including the PR description above) so we can discuss how these changes ought to work.

oyvindseland commented:
So the original commit should not change answers except for what is coming from the CAM update? Or does it not change the answers at all?

Commit:
Setndropmixed in chemistry.F90

Create list of Oslo Aero tracer names
Add ndropmixed property to Oslo Aero aerosol tracer constituents

gold2718 commented:
> So the original commit should not change answers except for what is coming from the CAM update? Or does it not change the answers at all?

Correct. I will open a CAM PR with details of the namelist and answer changes.

oyvindseland commented:
Looks fine and mostly technical updates. One question I see state has entered a number of subroutine calls, but as far as I can see not used. It is in preparation for secondary ice or local ice limiter? which does need state?

gold2718 commented:
> Looks fine and mostly technical updates. One question I see state has entered a number of subroutine calls, but as far as I can see not used. It is in preparation for secondary ice or local ice limiter? which does need state?

I think `state` has been added to provide a consistent interface across aerosol models. At the moment, I think the only model using this information is CARMA.
